src/tools/concurrency/TracingBackend.java

```diff
@@ -252,7 +264,9 @@ public static final void forceSwapBuffers() {
       }
     }
 
-    while (!fullBuffers.isEmpty()) {
+    // forceSwap should only happen for kompos.
```

Comment:
There assertion at the top of the method isn't exactly clear on that.
In Kompos, we force swap to be able to get results with good latency.

On shutdown, we probably also force swap, no? Which would be independent from Kompos.

Reply:
On Shutdown we have the onTermination methods in TracingActivityThread that returns the buffers normally.